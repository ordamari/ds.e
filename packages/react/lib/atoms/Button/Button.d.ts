import React from 'react';
interface ButtonProps {
    label: string;
}
declare function Button({ label }: ButtonProps): React.JSX.Element;
export default Button;
