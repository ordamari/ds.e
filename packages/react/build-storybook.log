@storybook/cli v7.0.27

info => Cleaning outputDir: C:\\Users\\User\\AppData\\Local\\Temp\\chromatic--26220-aXNlzomWoPln
info => Loading presets
info => Building manager..
info => Manager built (359 ms)
info => Compiling preview..
info => Copying static files: C:\Users\User\Desktop\My Projects\web\ds.e\node_modules\@storybook\manager\static at C:\Users\User\AppData\Local\Temp\chromatic--26220-aXNlzomWoPln\sb-common-assets
WARN You (or an addon) are using the 'config' preset field. This has been replaced by 'previewAnnotations' and will be removed in 8.0
info => Using implicit CSS loaders
info => Using default Webpack5 setup
<s> [webpack.Progress] 0% 

<s> [webpack.Progress] 1% setup before run
<s> [webpack.Progress] 1% setup before run NodeEnvironmentPlugin
<s> [webpack.Progress] 1% setup before run
<s> [webpack.Progress] 2% setup run
<s> [webpack.Progress] 2% setup run
<s> [webpack.Progress] 4% setup normal module factory
<s> [webpack.Progress] 4% setup normal module factory CaseSensitivePathsPlugin
<s> [webpack.Progress] 4% setup normal module factory
<s> [webpack.Progress] 5% setup context module factory
<s> [webpack.Progress] 5% setup context module factory
<s> [webpack.Progress] 6% setup before compile
<s> [webpack.Progress] 6% setup before compile ProgressPlugin
<s> [webpack.Progress] 6% setup before compile
<s> [webpack.Progress] 7% setup compile
<s> [webpack.Progress] 7% setup compile ExternalsPlugin
<s> [webpack.Progress] 7% setup compile ExternalsPlugin
<s> [webpack.Progress] 7% setup compile
<s> [webpack.Progress] 8% setup compilation
<s> [webpack.Progress] 8% setup compilation ArrayPushCallbackChunkFormatPlugin
<s> [webpack.Progress] 8% setup compilation JsonpChunkLoadingPlugin
<s> [webpack.Progress] 8% setup compilation StartupChunkDependenciesPlugin
<s> [webpack.Progress] 8% setup compilation ImportScriptsChunkLoadingPlugin
<s> [webpack.Progress] 8% setup compilation FetchCompileWasmPlugin
<s> [webpack.Progress] 8% setup compilation FetchCompileAsyncWasmPlugin
<s> [webpack.Progress] 8% setup compilation WorkerPlugin
<s> [webpack.Progress] 8% setup compilation SplitChunksPlugin
<s> [webpack.Progress] 8% setup compilation RuntimeChunkPlugin
<s> [webpack.Progress] 8% setup compilation ResolverCachePlugin
<s> [webpack.Progress] 8% setup compilation HtmlWebpackPlugin
<s> [webpack.Progress] 8% setup compilation
<s> [webpack.Progress] 9% setup compilation
<s> [webpack.Progress] 9% setup compilation DefinePlugin
<s> [webpack.Progress] 9% setup compilation ProvidePlugin
<s> [webpack.Progress] 9% setup compilation ProgressPlugin
<s> [webpack.Progress] 9% setup compilation ChunkPrefetchPreloadPlugin
<s> [webpack.Progress] 9% setup compilation SourceMapDevToolPlugin
<s> [webpack.Progress] 9% setup compilation JavascriptModulesPlugin
<s> [webpack.Progress] 9% setup compilation JsonModulesPlugin
<s> [webpack.Progress] 9% setup compilation AssetModulesPlugin
<s> [webpack.Progress] 9% setup compilation EntryPlugin
<s> [webpack.Progress] 9% setup compilation RuntimePlugin
<s> [webpack.Progress] 9% setup compilation InferAsyncModulesPlugin
<s> [webpack.Progress] 9% setup compilation DataUriPlugin
<s> [webpack.Progress] 9% setup compilation FileUriPlugin
<s> [webpack.Progress] 9% setup compilation CompatibilityPlugin
<s> [webpack.Progress] 9% setup compilation HarmonyModulesPlugin
<s> [webpack.Progress] 9% setup compilation AMDPlugin
<s> [webpack.Progress] 9% setup compilation RequireJsStuffPlugin
<s> [webpack.Progress] 9% setup compilation CommonJsPlugin
<s> [webpack.Progress] 9% setup compilation LoaderPlugin
<s> [webpack.Progress] 9% setup compilation LoaderPlugin
<s> [webpack.Progress] 9% setup compilation NodeStuffPlugin
<s> [webpack.Progress] 9% setup compilation APIPlugin
<s> [webpack.Progress] 9% setup compilation ExportsInfoApiPlugin
<s> [webpack.Progress] 9% setup compilation WebpackIsIncludedPlugin
<s> [webpack.Progress] 9% setup compilation ConstPlugin
<s> [webpack.Progress] 9% setup compilation UseStrictPlugin
<s> [webpack.Progress] 9% setup compilation RequireIncludePlugin
<s> [webpack.Progress] 9% setup compilation RequireEnsurePlugin
<s> [webpack.Progress] 9% setup compilation RequireContextPlugin
<s> [webpack.Progress] 9% setup compilation ImportPlugin
<s> [webpack.Progress] 9% setup compilation ImportMetaContextPlugin
<s> [webpack.Progress] 9% setup compilation SystemPlugin
<s> [webpack.Progress] 9% setup compilation ImportMetaPlugin
<s> [webpack.Progress] 9% setup compilation URLPlugin
<s> [webpack.Progress] 9% setup compilation DefaultStatsFactoryPlugin
<s> [webpack.Progress] 9% setup compilation DefaultStatsPresetPlugin
<s> [webpack.Progress] 9% setup compilation DefaultStatsPrinterPlugin
<s> [webpack.Progress] 9% setup compilation JavascriptMetaInfoPlugin
<s> [webpack.Progress] 9% setup compilation EnsureChunkConditionsPlugin
<s> [webpack.Progress] 9% setup compilation RemoveEmptyChunksPlugin
<s> [webpack.Progress] 9% setup compilation MergeDuplicateChunksPlugin
<s> [webpack.Progress] 9% setup compilation FlagIncludedChunksPlugin
<s> [webpack.Progress] 9% setup compilation SideEffectsFlagPlugin
<s> [webpack.Progress] 9% setup compilation FlagDependencyExportsPlugin
<s> [webpack.Progress] 9% setup compilation FlagDependencyUsagePlugin
<s> [webpack.Progress] 9% setup compilation InnerGraphPlugin
<s> [webpack.Progress] 9% setup compilation MangleExportsPlugin
<s> [webpack.Progress] 9% setup compilation ModuleConcatenationPlugin
<s> [webpack.Progress] 9% setup compilation NoEmitOnErrorsPlugin
<s> [webpack.Progress] 9% setup compilation RealContentHashPlugin
<s> [webpack.Progress] 9% setup compilation WasmFinalizeExportsPlugin
<s> [webpack.Progress] 9% setup compilation NamedModuleIdsPlugin
<s> [webpack.Progress] 9% setup compilation DeterministicChunkIdsPlugin
<s> [webpack.Progress] 9% setup compilation DefinePlugin
<s> [webpack.Progress] 9% setup compilation TerserPlugin
<s> [webpack.Progress] 9% setup compilation TemplatedPathPlugin
<s> [webpack.Progress] 9% setup compilation RecordIdsPlugin
<s> [webpack.Progress] 9% setup compilation WarnCaseSensitiveModulesPlugin
<s> [webpack.Progress] 9% setup compilation IgnoreWarningsPlugin
<s> [webpack.Progress] 9% setup compilation
<s> [webpack.Progress] 10% building
<s> [webpack.Progress] 10% building 0/1 entries 0/0 dependencies 0/0 modules
<s> [webpack.Progress] 26% building import loader ../../node_modules/babel-loader/lib/index.js
<s> [webpack.Progress] 26% building 0/1 entries 10/12 dependencies 3/10 modules
<s> [webpack.Progress] 26% building 0/1 entries 12/16 dependencies 3/10 modules
<s> [webpack.Progress] 29% building import loader ../../node_modules/@storybook/source-loader/dist/index.js
<s> [webpack.Progress] 29% building 0/1 entries 34/46 dependencies 9/26 modules
<s> [webpack.Progress] 29% building 0/1 entries 43/63 dependencies 10/28 modules
<s> [webpack.Progress] 44% building import loader ../../node_modules/style-loader/dist/cjs.js
<s> [webpack.Progress] 44% building import loader ../../node_modules/css-loader/dist/cjs.js
<s> [webpack.Progress] 44% building 0/1 entries 167/177 dependencies 57/90 modules
<s> [webpack.Progress] 65% building 1/1 entries 646/646 dependencies 266/266 modules
<s> [webpack.Progress] 65% building
<s> [webpack.Progress] 69% building finish
<s> [webpack.Progress] 69% building finish
<s> [webpack.Progress] 70% sealing finish module graph
<s> [webpack.Progress] 70% sealing finish module graph ResolverCachePlugin
<s> [webpack.Progress] 70% sealing finish module graph InferAsyncModulesPlugin
<s> [webpack.Progress] 70% sealing finish module graph FlagDependencyExportsPlugin
<s> [webpack.Progress] 70% sealing finish module graph InnerGraphPlugin
<s> [webpack.Progress] 70% sealing finish module graph WasmFinalizeExportsPlugin
<s> [webpack.Progress] 70% sealing finish module graph
<s> [webpack.Progress] 70% sealing plugins
<s> [webpack.Progress] 70% sealing plugins WarnCaseSensitiveModulesPlugin
<s> [webpack.Progress] 70% sealing plugins
<s> [webpack.Progress] 71% sealing dependencies optimization
<s> [webpack.Progress] 71% sealing dependencies optimization SideEffectsFlagPlugin
<s> [webpack.Progress] 71% sealing dependencies optimization FlagDependencyUsagePlugin
<s> [webpack.Progress] 71% sealing dependencies optimization
<s> [webpack.Progress] 71% sealing after dependencies optimization
<s> [webpack.Progress] 71% sealing after dependencies optimization
<s> [webpack.Progress] 72% sealing chunk graph
<s> [webpack.Progress] 72% sealing chunk graph
<s> [webpack.Progress] 73% sealing after chunk graph
<s> [webpack.Progress] 73% sealing after chunk graph
<s> [webpack.Progress] 73% sealing optimizing
<s> [webpack.Progress] 73% sealing optimizing
<s> [webpack.Progress] 74% sealing module optimization
<s> [webpack.Progress] 74% sealing module optimization
<s> [webpack.Progress] 75% sealing after module optimization
<s> [webpack.Progress] 75% sealing after module optimization
<s> [webpack.Progress] 75% sealing chunk optimization
<s> [webpack.Progress] 75% sealing chunk optimization EnsureChunkConditionsPlugin
<s> [webpack.Progress] 75% sealing chunk optimization RemoveEmptyChunksPlugin
<s> [webpack.Progress] 75% sealing chunk optimization MergeDuplicateChunksPlugin
<s> [webpack.Progress] 75% sealing chunk optimization SplitChunksPlugin
<s> [webpack.Progress] 75% sealing chunk optimization RemoveEmptyChunksPlugin
<s> [webpack.Progress] 75% sealing chunk optimization
<s> [webpack.Progress] 76% sealing after chunk optimization
<s> [webpack.Progress] 76% sealing after chunk optimization
<s> [webpack.Progress] 77% sealing module and chunk tree optimization
<s> [webpack.Progress] 77% sealing module and chunk tree optimization PersistentChildCompilerSingletonPlugin
<s> [webpack.Progress] 77% sealing module and chunk tree optimization
<s> [webpack.Progress] 77% sealing after module and chunk tree optimization
<s> [webpack.Progress] 77% sealing after module and chunk tree optimization
<s> [webpack.Progress] 78% sealing chunk modules optimization
<s> [webpack.Progress] 78% sealing chunk modules optimization ModuleConcatenationPlugin
<s> [webpack.Progress] 78% sealing chunk modules optimization
<s> [webpack.Progress] 78% sealing after chunk modules optimization
<s> [webpack.Progress] 78% sealing after chunk modules optimization
<s> [webpack.Progress] 79% sealing module reviving
<s> [webpack.Progress] 79% sealing module reviving RecordIdsPlugin
<s> [webpack.Progress] 79% sealing module reviving
<s> [webpack.Progress] 80% sealing before module ids
<s> [webpack.Progress] 80% sealing before module ids
<s> [webpack.Progress] 80% sealing module ids
<s> [webpack.Progress] 80% sealing module ids NamedModuleIdsPlugin
<s> [webpack.Progress] 80% sealing module ids
<s> [webpack.Progress] 81% sealing module id optimization
<s> [webpack.Progress] 81% sealing module id optimization
<s> [webpack.Progress] 82% sealing module id optimization
<s> [webpack.Progress] 82% sealing module id optimization
<s> [webpack.Progress] 82% sealing chunk reviving
<s> [webpack.Progress] 82% sealing chunk reviving RecordIdsPlugin
<s> [webpack.Progress] 82% sealing chunk reviving
<s> [webpack.Progress] 83% sealing before chunk ids
<s> [webpack.Progress] 83% sealing before chunk ids
<s> [webpack.Progress] 84% sealing chunk ids
<s> [webpack.Progress] 84% sealing chunk ids DeterministicChunkIdsPlugin
<s> [webpack.Progress] 84% sealing chunk ids
<s> [webpack.Progress] 84% sealing chunk id optimization
<s> [webpack.Progress] 84% sealing chunk id optimization FlagIncludedChunksPlugin
<s> [webpack.Progress] 84% sealing chunk id optimization
<s> [webpack.Progress] 85% sealing after chunk id optimization
<s> [webpack.Progress] 85% sealing after chunk id optimization
<s> [webpack.Progress] 86% sealing record modules
<s> [webpack.Progress] 86% sealing record modules RecordIdsPlugin
<s> [webpack.Progress] 86% sealing record modules
<s> [webpack.Progress] 86% sealing record chunks
<s> [webpack.Progress] 86% sealing record chunks RecordIdsPlugin
<s> [webpack.Progress] 86% sealing record chunks
<s> [webpack.Progress] 87% sealing module hashing
<s> [webpack.Progress] 87% sealing module hashing
<s> [webpack.Progress] 87% sealing code generation
<s> [webpack.Progress] 87% sealing code generation
<s> [webpack.Progress] 88% sealing runtime requirements
<s> [webpack.Progress] 88% sealing runtime requirements
<s> [webpack.Progress] 89% sealing hashing
<s> [webpack.Progress] 89% sealing hashing
<s> [webpack.Progress] 89% sealing after hashing
<s> [webpack.Progress] 89% sealing after hashing
<s> [webpack.Progress] 90% sealing record hash
<s> [webpack.Progress] 90% sealing record hash
<s> [webpack.Progress] 91% sealing module assets processing
<s> [webpack.Progress] 91% sealing module assets processing
<s> [webpack.Progress] 91% sealing chunk assets processing
<s> [webpack.Progress] 91% sealing chunk assets processing
<s> [webpack.Progress] 92% sealing asset processing
<s> [webpack.Progress] 92% sealing asset processing PersistentChildCompilerSingletonPlugin
<s> [webpack.Progress] 92% sealing asset processing TerserPlugin
<s> [webpack.Progress] 92% sealing asset processing SourceMapDevToolPlugin
<s> [webpack.Progress] 92% sealing asset processing SourceMapDevToolPlugin
<s> [webpack.Progress] 92% sealing asset processing SourceMapDevToolPlugin main.0d18829c.iframe.bundle.js generate SourceMap
<s> [webpack.Progress] 92% sealing asset processing SourceMapDevToolPlugin main.0d18829c.iframe.bundle.js generated SourceMap
<s> [webpack.Progress] 92% sealing asset processing SourceMapDevToolPlugin runtime~main.5b4aaf51.iframe.bundle.js generate SourceMap
<s> [webpack.Progress] 92% sealing asset processing SourceMapDevToolPlugin runtime~main.5b4aaf51.iframe.bundle.js generated SourceMap
<s> [webpack.Progress] 92% sealing asset processing SourceMapDevToolPlugin atoms-Color-Color-stories.11984c8a.iframe.bundle.js generate SourceMap
<s> [webpack.Progress] 92% sealing asset processing SourceMapDevToolPlugin atoms-Color-Color-stories.11984c8a.iframe.bundle.js generated SourceMap
<s> [webpack.Progress] 92% sealing asset processing SourceMapDevToolPlugin atoms-Text-Text-stories.e12744ec.iframe.bundle.js generate SourceMap
<s> [webpack.Progress] 92% sealing asset processing SourceMapDevToolPlugin atoms-Text-Text-stories.e12744ec.iframe.bundle.js generated SourceMap
<s> [webpack.Progress] 92% sealing asset processing SourceMapDevToolPlugin molecules-Select-Select-stories.d795d5d8.iframe.bundle.js generate SourceMap
<s> [webpack.Progress] 92% sealing asset processing SourceMapDevToolPlugin molecules-Select-Select-stories.d795d5d8.iframe.bundle.js generated SourceMap
<s> [webpack.Progress] 92% sealing asset processing SourceMapDevToolPlugin 892.4d988383.iframe.bundle.js generate SourceMap
<s> [webpack.Progress] 92% sealing asset processing SourceMapDevToolPlugin 892.4d988383.iframe.bundle.js generated SourceMap
<s> [webpack.Progress] 92% sealing asset processing SourceMapDevToolPlugin 434.f1188454.iframe.bundle.js generate SourceMap
<s> [webpack.Progress] 92% sealing asset processing SourceMapDevToolPlugin 434.f1188454.iframe.bundle.js generated SourceMap
<s> [webpack.Progress] 92% sealing asset processing SourceMapDevToolPlugin resolve sources
<s> [webpack.Progress] 92% sealing asset processing SourceMapDevToolPlugin
<s> [webpack.Progress] 92% sealing asset processing HtmlWebpackPlugin
<s> [webpack.Progress] 92% sealing asset processing HtmlWebpackPlugin
<s> [webpack.Progress] 92% sealing asset processing HtmlWebpackPlugin resolve sources
<s> [webpack.Progress] 92% sealing asset processing HtmlWebpackPlugin
<s> [webpack.Progress] 92% sealing asset processing RealContentHashPlugin
<s> [webpack.Progress] 92% sealing asset processing RealContentHashPlugin
<s> [webpack.Progress] 92% sealing asset processing RealContentHashPlugin main.32cd7a0c.iframe.bundle.js generate SourceMap
<s> [webpack.Progress] 92% sealing asset processing RealContentHashPlugin main.32cd7a0c.iframe.bundle.js generated SourceMap
<s> [webpack.Progress] 92% sealing asset processing RealContentHashPlugin runtime~main.f1f61d09.iframe.bundle.js generate SourceMap
<s> [webpack.Progress] 92% sealing asset processing RealContentHashPlugin runtime~main.f1f61d09.iframe.bundle.js generated SourceMap
<s> [webpack.Progress] 92% sealing asset processing RealContentHashPlugin atoms-Color-Color-stories.523d5590.iframe.bundle.js generate SourceMap
<s> [webpack.Progress] 92% sealing asset processing RealContentHashPlugin atoms-Color-Color-stories.523d5590.iframe.bundle.js generated SourceMap
<s> [webpack.Progress] 92% sealing asset processing RealContentHashPlugin atoms-Text-Text-stories.6838b6f6.iframe.bundle.js generate SourceMap
<s> [webpack.Progress] 92% sealing asset processing RealContentHashPlugin atoms-Text-Text-stories.6838b6f6.iframe.bundle.js generated SourceMap
<s> [webpack.Progress] 92% sealing asset processing RealContentHashPlugin molecules-Select-Select-stories.08182d6a.iframe.bundle.js generate SourceMap
<s> [webpack.Progress] 92% sealing asset processing RealContentHashPlugin molecules-Select-Select-stories.08182d6a.iframe.bundle.js generated SourceMap
<s> [webpack.Progress] 92% sealing asset processing RealContentHashPlugin 892.c5c511b7.iframe.bundle.js generate SourceMap
<s> [webpack.Progress] 92% sealing asset processing RealContentHashPlugin 892.c5c511b7.iframe.bundle.js generated SourceMap
<s> [webpack.Progress] 92% sealing asset processing RealContentHashPlugin 434.8a40debb.iframe.bundle.js generate SourceMap
<s> [webpack.Progress] 92% sealing asset processing RealContentHashPlugin 434.8a40debb.iframe.bundle.js generated SourceMap
<s> [webpack.Progress] 92% sealing asset processing RealContentHashPlugin resolve sources
<s> [webpack.Progress] 92% sealing asset processing RealContentHashPlugin
<s> [webpack.Progress] 92% sealing asset processing
<s> [webpack.Progress] 93% sealing after asset optimization
<s> [webpack.Progress] 93% sealing after asset optimization
<s> [webpack.Progress] 93% sealing recording
<s> [webpack.Progress] 93% sealing recording
<s> [webpack.Progress] 94% sealing after seal
<s> [webpack.Progress] 94% sealing after seal
<s> [webpack.Progress] 95% emitting emit
<s> [webpack.Progress] 95% emitting emit
<s> [webpack.Progress] 98% emitting after emit
<s> [webpack.Progress] 98% emitting after emit SizeLimitsPlugin
<s> [webpack.Progress] 98% emitting after emit
<s> [webpack.Progress] 99% done plugins
<s> [webpack.Progress] 99% done plugins CaseSensitivePathsPlugin
<s> [webpack.Progress] 99% done plugins
<s> [webpack.Progress] 99% 

<s> [webpack.Progress] 99% cache store build dependencies
<s> [webpack.Progress] 99% cache store build dependencies
<s> [webpack.Progress] 99% cache begin idle
<s> [webpack.Progress] 99% cache begin idle
<s> [webpack.Progress] 100% 

info => Preview built (18 s)
WARN asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
WARN This can impact web performance.
WARN Assets: 
WARN   892.c5c511b7.iframe.bundle.js (759 KiB)
WARN   434.8a40debb.iframe.bundle.js (683 KiB)
WARN entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
WARN Entrypoints:
WARN   main (694 KiB)
WARN       runtime~main.f1f61d09.iframe.bundle.js
WARN       434.8a40debb.iframe.bundle.js
WARN       main.32cd7a0c.iframe.bundle.js
WARN 
<s> [webpack.Progress] 99% cache shutdown
<s> [webpack.Progress] 99% cache shutdown
<s> [webpack.Progress] 100% 

info => Output directory: C:\\Users\\User\\AppData\\Local\\Temp\\chromatic--26220-aXNlzomWoPln
