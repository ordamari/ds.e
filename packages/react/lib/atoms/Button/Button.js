import React from 'react';

function Button({ label }) {
    return React.createElement("button", { className: 'dse-button-container' }, label);
}

export { Button as default };
//# sourceMappingURL=Button.js.map
