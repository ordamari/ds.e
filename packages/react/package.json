{
  "name": "@or.ds.e/react",
  "version": "0.5.0",
  "main": "lib/index.js",
  "license": "MIT",
  "scripts": {
    "build": "rollup -c --bundleConfigAsCjs",
    "dev": "yarn build --watch",
    "test": "jest --verbose ",
    "test:watch": "yarn test --watch",
    "start:storybook": "storybook dev",
    "build:storybook": "storybook build",
    "build-storybook": "storybook build",
    "test:chromatic": "chromatic --project-token=\"$CHROMATIC_PROJECT_TOKEN\""
  },
  "devDependencies": {
    "@babel/preset-env": "^7.22.9",
    "@babel/preset-react": "^7.22.5",
    "@babel/preset-typescript": "^7.22.5",
    "@or.ds.e/foundation": "^0.5.0",
    "@or.ds.e/helpers": "^0.5.0",
    "@storybook/addon-a11y": "^7.0.27",
    "@storybook/addon-essentials": "7.0.27",
    "@storybook/addon-interactions": "7.0.27",
    "@storybook/addon-knobs": "^7.0.2",
    "@storybook/addon-links": "7.0.27",
    "@storybook/addon-storysource": "^7.0.27",
    "@storybook/blocks": "7.0.27",
    "@storybook/preset-typescript": "^3.0.0",
    "@storybook/react": "^7.0.27",
    "@storybook/react-vite": "7.0.27",
    "@storybook/react-webpack5": "7.0.27",
    "@storybook/testing-library": "0.0.14-next.2",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^14.0.0",
    "@types/jest": "^29.5.3",
    "@types/react": "^18.2.14",
    "chromatic": "^6.19.9",
    "jest": "^29.6.1",
    "jest-environment-jsdom": "^29.6.1",
    "prop-types": "15.8.1",
    "react": "^18.2.0",
    "react-dom": "18.2.0",
    "rollup": "^3.26.2",
    "rollup-plugin-typescript2": "^0.35.0",
    "storybook": "7.0.27",
    "typescript": "^5.1.6"
  },
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "lib"
  ],
  "gitHead": "226830b43cb46c199f0ca65430bcff1082467934"
}
