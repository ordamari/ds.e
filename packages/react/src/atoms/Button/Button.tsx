import React from 'react';

interface ButtonProps {
    label: string;
}

function Button({label}: ButtonProps){
    return <button
    className='dse-button-container'
    >{label}</button>
}

export default Button;