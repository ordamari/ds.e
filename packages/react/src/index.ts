import Color from './atoms/Color';
import Text from './atoms/Text';
import Margin from './atoms/Margin';
import Icon from './atoms/Icon';

import Select from './molecules/Select';
import Button from './molecules/Button';
import Input from './molecules/Input';

import useToggle from './hooks/useToggle';
import useFocusWithin from './hooks/useFocusWithin';
export {
  Color,
  Text,
  Margin,
  Icon,
  Select,
  Button,
  Input,
  useToggle,
  useFocusWithin,
};
