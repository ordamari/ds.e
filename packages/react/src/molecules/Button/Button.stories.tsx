import React from 'react';
import Button from './Button';
import { select, boolean, text } from '@storybook/addon-knobs';
import '@or.ds.e/scss/lib/Button.css';
import '@or.ds.e/scss/lib/Utilities.css';
import { Variant, FontSize, Icon } from '@or.ds.e/foundation';

const variantOptions = Object.keys(Variant) as Array<keyof typeof Variant>;
const sizeOptions = Object.keys(FontSize) as Array<keyof typeof FontSize>;
const iconOptions = Object.keys(Icon) as Array<keyof typeof Icon>;

export default {
  title: 'Molecules/Button',
};

export const Common = () => <Button>{text('Label', 'Button')}</Button>;

export const Variants = () => (
  <Button variant={select('Variant', variantOptions, Variant.primary)}>
    Button
  </Button>
);

export const Sizes = () => (
  <Button size={select('Size', sizeOptions, FontSize.base)}>Button</Button>
);

export const Disabled = () => (
  <Button
    variant={select('Variant', variantOptions, Variant.primary)}
    disabled={boolean('Disabled', true)}
  >
    Button
  </Button>
);

export const WithIcon = () => (
  <Button icon={select('Icon', iconOptions, Icon.caret)}>Button</Button>
);
