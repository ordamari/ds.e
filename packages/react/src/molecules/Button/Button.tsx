import React from 'react';
import { Variant, FontSize, Icon as IconOptions } from '@or.ds.e/foundation';
import Text from '../../atoms/Text';
import Icon from '../../atoms/Icon';
interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  children: React.ReactNode;
  variant?: keyof typeof Variant;
  size?: keyof typeof FontSize;
  icon?: keyof typeof IconOptions;
}

const Button = ({
  children,
  className: customClassName = '',
  size = FontSize.base,
  variant = Variant.primary,
  icon,
  ...props
}: ButtonProps) => {
  const className = `${customClassName} dse-button dse-button--${variant}`;
  return (
    <button className={className} {...props}>
      {icon ? <Icon className="dse-button__icon" icon={icon} /> : null}
      <Text size={size}>{children}</Text>
    </button>
  );
};

export default Button;
