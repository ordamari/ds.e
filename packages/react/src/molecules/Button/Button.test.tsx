import React from 'react';
import Button from './Button';
import { render, fireEvent } from '@testing-library/react';

test('renders the label passed to it', () => {
  const { getByText } = render(<Button>Button</Button>);

  expect(getByText('Button')).toBeInTheDocument();
});

test('calls the onClick prop when clicked', () => {
  const onClick = jest.fn();
  const { getByText } = render(<Button onClick={onClick}>Button</Button>);
  fireEvent.click(getByText('Button'));
  expect(onClick).toHaveBeenCalled();
});

test('renders the icon passed to it', () => {
  const { getByTestId } = render(<Button icon="caret">Button</Button>);
  expect(getByTestId('DseIcon')).toBeInTheDocument();
});

test('Does not render the icon if no icon prop is passed', () => {
  const { queryByTestId } = render(<Button>Button</Button>);
  expect(queryByTestId('DseIcon')).not.toBeInTheDocument();
});

test('Disables the button if the disabled prop is passed', () => {
  const onClick = jest.fn();
  const { getByText } = render(
    <Button onClick={onClick} disabled>
      Button
    </Button>,
  );
  const button = getByText('Button');
  fireEvent.click(button);
  expect(onClick).not.toHaveBeenCalled();
});

test('Snapshot primary', () => {
  const { container } = render(<Button>Button</Button>);
  expect(container).toMatchSnapshot();
});

test('Snapshot with icon', () => {
  const { container } = render(<Button icon="caret">Button</Button>);
  expect(container).toMatchSnapshot();
});
