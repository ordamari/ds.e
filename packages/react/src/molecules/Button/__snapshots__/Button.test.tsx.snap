// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot primary 1`] = `
<div>
  <button
    class=" dse-button dse-button--primary"
  >
    <p
      class="dse-text dse-text-base"
    >
      Button
    </p>
  </button>
</div>
`;

exports[`Snapshot with icon 1`] = `
<div>
  <button
    class=" dse-button dse-button--primary"
  >
    <svg
      aria-hidden="true"
      class=" dse-width-sm dse-height-sm"
      data-testid="DseIcon"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19.5 8.25l-7.5 7.5-7.5-7.5"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
    <p
      class="dse-text dse-text-base"
    >
      Button
    </p>
  </button>
</div>
`;
